packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/state_var.rs

```diff
@@ -94,50 +92,47 @@ pub enum RequestDependencyUpdateError {
     CouldNotUpdate,
 }
 
-/// StateVarInterface defines the relationship between the state variable
-/// and its dependencies
-pub trait StateVarInterface<T: Default + Clone>: std::fmt::Debug {
-    /// Return the instructions that can be used to calculate the dependencies
-    /// for this state variable based on the structure of the document,
+/// Methods used when updating a state variable's dependencies, including querying for its
+/// dependencies and calculating the value from its dependencies.
+pub trait StateVarUpdaters<T: Default + Clone>: std::fmt::Debug {
+    /// Returns the data queries needed to calculate the dependencies
+    /// for this state variable. These queries may be based on structure of the document,
     /// e.g., the children, attributes, or other state variables
-    /// of the component of this state variable
-    fn return_dependency_instructions(
+    /// of the component of this state variable.
+    fn return_data_queries(
         &mut self,
         extending: Option<ExtendSource>,
         state_var_idx: StateVarIdx,
-    ) -> Vec<DependencyInstruction>;
+    ) -> Vec<DataQuery>;
 
-    /// Given the structure of the document and the dependency instructions,
-    /// the actual dependencies will be determined and passed to `save_dependencies`.
-    /// The function `save_dependencies` should store
-    /// the dependencies directly on the the structure (`self`)
-    /// in a form (presumably typed not with enums) for efficient calculation.
+    /// Called when data queries for the state variable have been completed.
+    /// State variables are expected to cache the results of their queries
```

Comment:
Maybe "State variables must cache...", as "expected to" sounds like it might be optional to me.

Reply:
I guess the types might be hard to figure out, but this bit bothered me...it seems like core should cache the results and send them to the component when they're needed...

Reply:
Yeah, in the Javascript code, core does the caching. I did this solely for the typing, to avoid unwrapping enums on every cycle. Maybe a silly optimization that we can revisit with profiling.